include/riak_kv_sweeper.hrl

```diff
@@ -0,0 +1,65 @@
+%% Copyright (c) 2011-2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+
+%% Used as configuration for sweep participants.
+-record(sweep_participant,
+        {
+         description :: string(),   %% Human readeble description of the user.
+         module :: atom(),          %% module where the sweep call back lives.
+         fun_type :: riak_kv_sweeper:fun_type(), %% delete_fun | modify_fun | observe_fun
+         sweep_fun :: fun(),        %%
+         run_interval :: integer() | fun(), %% Defines how often participant wants to run.
+         acc :: any(),
+         options = [] :: list(),    %% optional values that will be added during sweep
+         errors = 0 :: integer(),
+         fail_reason
+        }).
+
+-record(sweep,
```

Comment:
For `#sweep{}` the changes are not that big, so I think we should do it now rather than later.

Outside `riak_kv_sweeper_state` `#sweep{}` is only used here (disregarding the test):
```
src/riak_kv_sweeper.erl
139:-spec status() -> {[#sweep_participant{}], [#sweep{}]}.

src/riak_kv_sweeper_cli.erl
100:              || #sweep{state = State} = Sweep <- Sweeps, State == running],
110:format_progress(#sweep{index = Index,

src/riak_kv_sweeper_fold.erl
284:                           #sweep{}) -> 'ok' | 'no_pid'.
285:send_to_sweep_worker(Msg, #sweep{pid = Pid}) when is_pid(Pid) ->
289:send_to_sweep_worker(Msg, #sweep{index = Index}) ->
```

`riak_kv_sweeper` is just a type definition.

`riak_kv_sweeper_cli`: first usage can be fixed by adding a function to extract the state.
2nd usage would require moving the formatting into the `riak_kv_sweeper_state` module, which is not unfair given that it is relating to a type - quite a common pattern in OCaml.

`riak_kv_sweeper_fold`: accessor functions to get the pid and index out.

For this record it makes sense to hide it.